src/refda/adm/ietf_dtnma_agent.c

```diff
@@ -1874,8 +1880,8 @@ static void refda_adm_ietf_dtnma_agent_ctrl_ensure_odm(refda_ctrl_exec_ctx_t *ct
         return;
     }
 
-    org_name = CACE_MALLOC(strlen(org->ptr) + 1);
-    strcpy(org_name, org->ptr);
+    org_name = CACE_MALLOC(strlen((char *)org->ptr) + 1);
+    strcpy(org_name, (char *)org->ptr);
```

Comment:
Some better bookkeeping of these allocations is needed to avoid opportunity for leaks. The main container can define an object pool for `m_string_t` values for ODM names. Let me know if you want a pointer about doing this.

Reply:
Updated the code to prevent memory leaks from these allocations within `refda_adm_ietf_dtnma_agent_ctrl_ensure_odm`. Let me know if you still recommend the object pool approach.